clang/lib/CIR/CodeGen/CIRGenExpr.cpp

```diff
@@ -1145,10 +1145,6 @@ RValue CIRGenFunction::buildCall(clang::QualType CalleeType,
   if (isa<FunctionNoProtoType>(FnType) || Chain) {
     assert(!UnimplementedFeature::chainCalls());
     assert(!UnimplementedFeature::addressSpace());
-
-    // Set no-proto function as callee.
-    auto Fn = llvm::dyn_cast<mlir::cir::FuncOp>(Callee.getFunctionPointer());
-    Callee.setFunctionPointer(Fn);
```

Comment:
Not sure this is the right fix, why removing the code instead of passing down `Callee.getFunctionPointer()` either directly or after a bitcast?

Reply:
We don't need to directly pass `Callee.getFunctionPointer()` because the code like `Callee.setFunctionPointer(Callee.getFunctionPointer())` is just sets fnction pointer to the same value.

About bitcast. I suppose that `bitcast` in original codegen is fully redundant becasue of it casts the 'CalleePtr' to the same type that `CalleePtr` already have. And the comment above doesn't describe the things that happens in code. 
https://github.com/llvm/clangir/blob/b27ad0fce5207093d966be2ccecf7164c9ff2e05/clang/lib/CodeGen/CGExpr.cpp#L5874-L5902
Moreover, unfortunately I can't find any case then this bitcast is necessary in original codegen.
So I assume that we shouldn't transfer emitting the bitcast to our CIR codegen.

Reply:
> We don't need to directly pass Callee.getFunctionPointer() because the code like Callee.setFunctionPointer(Callee.getFunctionPointer()) is just sets fnction pointer to the same value.

yea, my bad, just realized it sets it again because of the bitcast! 

> Moreover, unfortunately I can't find any case then this bitcast is necessary in original codegen.

To that intent, please add an assert for when `CalleePtr.getType() != CalleeTy`, so we can track whenever it shows up.

Reply:
Sorry, looks like the bitcast isn't redundant. I added bitcast to non-variadic function type in my PR.